problem_builder/mentoring.py

```diff
@@ -776,3 +789,98 @@ def workbench_scenarios():
+        """
+        from .step import MentoringStepBlock  # Import here to avoid circular dependency
+        return [
+            _normalize_id(child_id) for child_id in self.children if
+            child_isinstance(self, child_id, MentoringStepBlock)
+        ]
+
+    def student_view(self, context):
+        fragment = Fragment()
+        child_content = u""
+
+        for child_id in self.children:
+            child = self.runtime.get_block(child_id)
+            if child is None:  # child should not be None but it can happen due to bugs or permission issues
+                child_content += u"<p>[{}]</p>".format(self._(u"Error: Unable to load child component."))
+            elif not isinstance(child, MentoringMessageBlock):
+                try:
+                    child_fragment = child.render('mentoring_view', context)
+                except NoSuchViewError:
+                    if child.scope_ids.block_type == 'html' and getattr(self.runtime, 'is_author_mode', False):
+                        # html block doesn't support mentoring_view, and if we use student_view Studio will wrap
+                        # it in HTML that we don't want in the preview. So just render its HTML directly:
+                        child_fragment = Fragment(child.data)
+                    else:
+                        child_fragment = child.render('student_view', context)
+                fragment.add_frag_resources(child_fragment)
+                child_content += child_fragment.content
+
+        fragment.add_content(loader.render_template('templates/html/mentoring.html', {
+            'self': self,
+            'title': self.display_name,
+            'show_title': self.show_title,
+            'child_content': child_content,
+        }))
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/problem-builder.css'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/vendor/underscore-min.js'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/util.js'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/mentoring_standard_view.js'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/mentoring.js'))
+        fragment.add_resource(loader.load_unicode('templates/html/mentoring_attempts.html'), "text/html")
+        fragment.add_resource(loader.load_unicode('templates/html/mentoring_grade.html'), "text/html")
+        fragment.add_resource(loader.load_unicode('templates/html/mentoring_review_questions.html'), "text/html")
+
+        self.include_theme_files(fragment)
+        # Workbench doesn't have font awesome, so add it:
+        if WorkbenchRuntime and isinstance(self.runtime, WorkbenchRuntime):
+            fragment.add_css_url('//maxcdn.bootstrapcdn.com/font-awesome/4.3.0/css/font-awesome.min.css')
+
+        fragment.initialize_js('MentoringBlock')
+
+        return fragment
```

Comment:
@itsjeyd this student view is overly complicated - it is copy-pasted from MentoringXBlock, and we don't need that much stuff here. We have a dedicated task to make everything work in LMS, but if it is used in Studio as a preview it would make sense to get it somewhat working now. Alternative approach would be to provide a dedicated preview_view and make studio use it for preview.